src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -2,581 +2,428 @@
-            p.EnableRaisingEvents = true;
-            p.Exited += p_Exited;
-            p.Start();
-            p.Kill();
-            p.WaitForExit();
-            Assert.True(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0001: {0}", "isExited Event not called when EnableRaisingEvent is set to true."));
-
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
-
-            // Check with the default settings (false, events will not be raised)
-            p.Refresh();
-            p = CreateProcessInfinite();
-            p.Exited += p_Exited;
-            p.Start();
-            p.Kill();
-            p.WaitForExit();
-            Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0002: {0}", "isExited Event called with the default settings for EnableRaiseEvents"));
-
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
-
-            // Same test, this time explicitly set the property to false
-            p = CreateProcessInfinite();
-            p.EnableRaisingEvents = false;
-            p.Exited += p_Exited;
-            p.Start();
-            p.Kill();
-            p.WaitForExit();
-            Assert.False(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0003: {0}", "isExited Event called with the EnableRaiseEvents = false"));
-
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+            Task.Delay(TimeSpan.FromMilliseconds(delayInMs)).Wait();
         }
 
-        static void p_Exited(object sender, EventArgs e)
+        public void DelayTask()
         {
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = true;
+            DelayTask(50D);
         }
 
-        [Fact]
-        public static void Process_ExitCode()
+        public void StartAndKillProcessWithDelay(Process p)
         {
-            Process p = CreateProcess();
-            p.Start();
-            p.WaitForExit();
-            Assert.Equal(p.ExitCode, 100);
-
-            p = CreateProcessInfinite();
             p.Start();
+            DelayTask();
             p.Kill();
             p.WaitForExit();
-            Assert.True(p.ExitCode < 0, String.Format("Process_ExitCode: Unexpected Exit code {0}", p.ExitCode));
         }
 
         [Fact]
-        public static void Process_ExitTime()
+        public void Process_EnableRaiseEvents()
         {
-            Process p = CreateProcessInfinite();
-            p.Start();
-            p.Kill();
-            p.WaitForExit();
-            DateTime exitTime = p.ExitTime.ToUniversalTime();
-            DateTime dt2 = DateTime.UtcNow;
-            TimeSpan elapsedTime = new TimeSpan(dt2.Ticks - exitTime.Ticks);
-            Assert.True(elapsedTime.TotalSeconds <= 1, "Process_ExitTime is incorrect.");
-        }
+            bool isExitedInvoked = false;
```

Comment:
Thanks for making this change to use a local.  As long as you've gone this far, and since you're already resetting it to false inside each of the scopes, how about defining it instead inside each of the scopes, i.e. below instead of doing:
```
{
    isExitedInvoked = false;
    ...
}
```
do:
```
{
    bool isExitedInvoked = false;
    ...
}
```
The problem I'm trying to avoid here is one I've hit in the past with other threading tests: you capture something into a lambda, and then inadvertently one former test ends up unexpectedly affecting a latter test.  If each test uses its own bool, that won't happen.  By changing it from a static to a local, that shrunk the scope, and by moving it to each individual scope, we shrink the scope further.